lib/uplink/scope.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/gogo/protobuf/proto"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+)
+
+// Scope is a serializable type that represents all of the credentials you need
+// to open a project and some amount of buckets
+type Scope struct {
+	SatelliteAddr string
+
+	APIKey APIKey
+
+	EncryptionAccess *EncryptionAccess
```

Comment:
I cannot recall, what was the reason we are not combining APIKey and EncryptionAccess?

Reply:
api keys are sent to the satellite, and encryptionaccesses are purely client-side. api keys control what operations the satellite will allow, encryptionaccesses are private key material that you don't want to share with anyone you don't have to (including the satellite)